src/clients/integration.zig

```diff
@@ -5,83 +5,56 @@
 //
 
 const std = @import("std");
 const builtin = @import("builtin");
+const assert = std.debug.assert;
+
+const flags = @import("../flags.zig");
 
-const java_docs = @import("./java/docs.zig").JavaDocs;
-const dotnet_docs = @import("./dotnet/docs.zig").DotnetDocs;
-const go_docs = @import("./go/docs.zig").GoDocs;
-const node_docs = @import("./node/docs.zig").NodeDocs;
 const Docs = @import("./docs_types.zig").Docs;
 const TmpDir = @import("./shutil.zig").TmpDir;
 const git_root = @import("./shutil.zig").git_root;
 const run_shell = @import("./shutil.zig").run_shell;
 const prepare_directory = @import("./docs_generate.zig").prepare_directory;
 const integrate = @import("./docs_generate.zig").integrate;
 
+const CliArgs = struct {
+    language: enum { java, node, go, dotnet },
+    sample: []const u8,
+    keep_tmp: bool = false,
+};
+
 fn error_main() !void {
     var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
     defer arena.deinit();
 
     const allocator = arena.allocator();
 
+    var args = try std.process.argsWithAllocator(allocator);
+    defer args.deinit();
+
+    // Skip the first argument of the exe.
+    assert(args.skip());
+
+    const cli_args = flags.parse_flags(&args, CliArgs);
+
     const root = try git_root(&arena);
 
     std.debug.print("Moved to git root: {s}\n", .{root});
     try std.os.chdir(root);
 
-    var args = try std.process.argsWithAllocator(allocator);
-    defer args.deinit();
-
-    // Skip the first argument of the exe.
-    std.debug.assert(args.skip());
-
-    var language: ?Docs = null;
-    var sample: []const u8 = "";
-    var keep_tmp = false;
-    while (args.next()) |arg| {
-        if (std.mem.eql(u8, arg, "--language")) {
-            const next = args.next() orelse "";
-
-            if (std.mem.eql(u8, next, "java")) {
-                language = java_docs;
-            } else if (std.mem.eql(u8, next, "node")) {
-                language = node_docs;
-            } else if (std.mem.eql(u8, next, "go")) {
-                language = go_docs;
-            } else if (std.mem.eql(u8, next, "dotnet")) {
-                language = dotnet_docs;
-            } else {
-                std.debug.print("Unknown language: {s}.\n", .{next});
-                return error.UnknownLanguage;
-            }
-        }
-
-        if (std.mem.eql(u8, arg, "--keep-tmp")) {
-            keep_tmp = true;
-        }
-
-        if (std.mem.eql(u8, arg, "--sample")) {
-            const next = args.next() orelse "";
-
-            sample = next;
-        }
-    }
-
-    if (sample.len == 0) {
-        std.debug.print("--sample not set.\n", .{});
-        return error.SampleNotSet;
-    }
-
-    if (language == null) {
-        std.debug.print("--language not set.\n", .{});
-        return error.LanguageNotSet;
-    }
+    const language: Docs = switch (cli_args.language) {
+        .java => @import("./java/docs.zig").JavaDocs,
```

Comment:
Do we not have a policy of keeping imports at the top of the file for readability? It seems weird to me, even though these aren't dynamic imports, that we "hide" the imports until down here.

Reply:
Good question, I don't know! My guess would be that we would, in general, prefer to keep imports on top, but this isn't really an import of code, it's just a table with data, so its significantly less noisy to not repeat it in two places.

Reply:
There is actually code in them too though, the hooks anywawy.

Reply:
Raised discussion point in Slack. I don't feel too strongly here, but on balance compactness wins here for me. 

I was toying with the idea of defining 

```zig
pub const docs_for = .{
     .java = @import("./java/docs.zig"),
}
```

in the `doc_types.zig` so that we have a single source of truth there, instead of integration.zig and docs_generate repeating each other, but I don't like how this introduces circular dependencies.